kubectl-rook-ceph.sh

```diff
@@ -245,6 +249,20 @@ function run_rook_cr_status() {
   fi
 }
 
+function run_purge_osd() {
+  force_removal=false
+  if [ "$#" -eq 2 ] && [ "$2" = "--force" ]; then
+    force_removal=true
+  fi
+  mon_endpoints=$($TOP_LEVEL_COMMAND --namespace "$ROOK_CLUSTER_NAMESPACE" get cm rook-ceph-mon-endpoints -o jsonpath='{.data.data}' | cut -d "," -f1)
+  ceph_secret=$($TOP_LEVEL_COMMAND --namespace "$ROOK_OPERATOR_NAMESPACE" exec deploy/rook-ceph-operator -- cat /var/lib/rook/"$ROOK_CLUSTER_NAMESPACE"/client.admin.keyring | grep "key" | awk '{print $3}')
+  $TOP_LEVEL_COMMAND --namespace "$ROOK_OPERATOR_NAMESPACE" exec deploy/rook-ceph-operator -- sh -c "export ROOK_MON_ENDPOINTS=$mon_endpoints \
+      ROOK_CEPH_USERNAME=client.admin \
```

Comment:
Of note, the `rook-ceph-mon` secret has the name of the client used to access the cluster. Why don't we read that value and use it. Then this should be true now and in the future also.

Reply:
I also wonder if it wouldn't be better to start a version of the osd removal job, then tail the logs of that, instead of running the command in the operator. @travisn wdyt?

Reply:
> Of note, the `rook-ceph-mon` secret has the name of the client used to access the cluster. Why don't we read that value and use it. Then this should be true now and in the future also. It would also better support running actions on external clusters.

In the future there will basically be multiple users in the secret instead of just a single user. We will still have to choose which user, so not sure there's an advantage of reading it.

> I also wonder if it wouldn't be better to start a version of the osd removal job, then tail the logs of that, instead of running the command in the operator since it seems like that is causing us to have to do some pretty deep inspection of resources. @travisn wdyt?

The osd removal job is just running a single ceph command, which is also what we're doing with the plugin in the operator pod. Since the pattern of the plugin is to run in the operator pod, I don't see a benefit to changing that pattern. We would just change the complexity of reading keys for the complexity of starting a job and tailing its logs, right?